src/Splits.sol

```diff
@@ -240,15 +244,16 @@ abstract contract Splits {
     /// @notice Validates a list of splits receivers and emits events for them
     /// @param receivers The list of splits receivers
     /// Must be sorted by the account IDs, without duplicate account IDs and without 0 weights.
-    function _assertSplitsValid(SplitsReceiver[] memory receivers) private pure {
+    function _assertSplitsValid(SplitsReceiver[] calldata receivers) private pure {
         unchecked {
             require(receivers.length <= _MAX_SPLITS_RECEIVERS, "Too many splits receivers");
             uint256 totalWeight = 0;
             uint256 prevAccountId = 0;
             for (uint256 i = 0; i < receivers.length; i++) {
-                SplitsReceiver memory receiver = receivers[i];
-                uint32 weight = receiver.weight;
+                SplitsReceiver calldata receiver = receivers[i];
+                uint256 weight = receiver.weight;
                 require(weight != 0, "Splits receiver weight is zero");
+                if (weight > _TOTAL_SPLITS_WEIGHT) weight = _TOTAL_SPLITS_WEIGHT + 1;
```

Comment:
Okay, just to verify if I understood this correctly. 
```solidity
weight = _TOTAL_SPLITS_WEIGHT + 1;
```

This is a gas optimization. You want to operate with `unchecked` and still have a  the potential overflow protection.  Therefore you check if the `weight` is greater than the `_TOTAL_SPLITS_WEIGHT`. 

You add `+1` to ensure it will revert later with `totalWeight <= _TOTAL_SPLITS_WEIGHT`  an additional `require` would be more expensive.

Reply:
Yes, exactly :+1: